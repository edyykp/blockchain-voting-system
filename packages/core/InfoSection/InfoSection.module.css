.container {
  color: var(--white);
  width: 100%;
}

.primaryContainer {
  background: var(--white);
}

.secondaryContainer {
  background: var(--gray-800);
}

.wrapper {
  display: grid;
  z-index: 1;
  width: 100%;
  height: 90vh;
  max-width: 1100px;
  margin-right: auto;
  margin-left: auto;
  padding: 0 24px;
  justify-content: center;
}

.row {
  display: grid;
  grid-auto-columns: minmax(auto, 1fr);
  align-items: center;
  gap: 10vw;
}

.column1 {
  margin-bottom: 15px;
  padding: 0 15px;
  grid-area: col1;
}

.column2 {
  margin-bottom: 15px;
  padding: 0 15px;
  grid-area: col2;
}

.primaryRow {
  grid-template-areas: 'col2 col1';
}

.secondaryRow {
  grid-template-areas: 'col1 col2';
}

.textWrapper {
  max-width: 540px;
  padding-top: 0;
  padding-bottom: 60px;
}

.topLine {
  color: var(--primary);
  font-size: 16px;
  line-height: 16px;
  font-weight: 700;
  letter-spacing: 1.4px;
  text-transform: uppercase;
  margin-bottom: 16px;
}

.heading {
  margin-bottom: 24px;
  font-size: 48px;
  line-height: 1.1;
  font-weight: 600;
}

.primaryHeading {
  color: var(--gray-700);
}

.secondaryHeading {
  color: var(--gray-200);
}

.subtitle {
  max-width: 440px;
  margin-bottom: 35px;
  font-size: 18px;
  line-height: 24px;
}

.primarySubtitle {
  color: var(--gray-900);
}

.secondarySubtitle {
  color: var(--primary);
}

.btnWrap {
  display: flex;
  justify-content: flex-start;
}

.imgWrap {
  max-width: 555px;
  height: 100%;
}

.img {
  width: 100%;
  margin: 0 0 10px 0;
  padding-right: 0;
}

@media screen and (max-width: 768px) {
  .container {
    padding: 100px 0;
  }

  .primaryRow {
    grid-template-areas: 'col1' 'col2';
  }

  .secondaryRow {
    grid-template-areas: 'col1 col1' 'col2 col2';
  }
}

@media screen and (max-width: 480px) {
  .heading {
    font-size: 32px;
  }
}
