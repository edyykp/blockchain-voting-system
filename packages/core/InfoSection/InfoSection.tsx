import { Button } from '@packages/components';
import styles from './InfoSection.module.css';

type InfoSectionProps = {
  theme: 'primary' | 'secondary';
  svgPath: string;
  subtitle: string;
  heading: string;
  topline: string;
};
export const InfoSection = ({
  theme,
  svgPath,
  subtitle,
  heading,
  topline,
}: InfoSectionProps) => {
  return (
    <div
      className={`${styles.container} ${
        theme === 'primary'
          ? styles.primaryContainer
          : styles.secondaryContainer
      }`}
    >
      <div className={styles.wrapper}>
        <div
          className={`${styles.row} ${
            theme === 'primary' ? styles.primaryRow : styles.secondaryRow
          }`}
        >
          <div className={styles.column1} data-aos="fade-up">
            <div className={styles.textWrapper}>
              <p className={styles.topLine}>{topline}</p>
              <h1
                className={`${styles.heading} ${
                  theme === 'primary'
                    ? styles.primaryHeading
                    : styles.secondaryHeading
                }`}
              >
                {heading}
              </h1>
              <p
                className={`${styles.subtitle} ${
                  theme === 'primary'
                    ? styles.primarySubtitle
                    : styles.secondarySubtitle
                }`}
              >
                {subtitle}
              </p>
              <div className={styles.btnWrap}>
                {theme === 'primary' && <Button theme="primary" size="md" />}
                {theme === 'secondary' && (
                  <Button theme="secondary" size="md" />
                )}
              </div>
            </div>
          </div>
          <div className={styles.column2} data-aos="fade-up">
            <div className={styles.imgWrap}>
              <img
                className={styles.img}
                alt="info-section-img"
                src={svgPath}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
