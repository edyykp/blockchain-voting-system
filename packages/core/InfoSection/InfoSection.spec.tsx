import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { InfoSection } from './InfoSection';

const mockData = {
  heading: 'heading',
  subtitle: 'subtitle',
  topline: 'topline',
  svgPath: '/svg.svg',
  id: 'about',
  href: 'https://www.google.com',
  buttonText: 'Find out more',
};

const setup = (theme: 'primary' | 'secondary') => {
  return {
    ...render(<InfoSection {...mockData} theme={theme} />),
  };
};

describe('InfoSectionâ†’', () => {
  it('renders', () => {
    const { container, getByTestId } = setup('primary');

    expect(getByTestId('info-section')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
