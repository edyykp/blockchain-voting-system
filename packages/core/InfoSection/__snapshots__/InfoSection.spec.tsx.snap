// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoSectionâ†’ renders 1`] = `
<div>
  <div
    class="container primaryContainer"
    data-testid="info-section"
    id="about"
  >
    <div
      class="wrapper"
    >
      <div
        class="row primaryRow"
      >
        <div
          class="column1"
          data-aos="fade-up"
        >
          <div
            class="textWrapper"
          >
            <p
              class="topLine"
            >
              topline
            </p>
            <h1
              class="heading primaryHeading"
            >
              heading
            </h1>
            <p
              class="subtitle primarySubtitle"
            >
              subtitle
            </p>
            <div
              class="btnWrap"
              data-testid="btn-wrap"
            >
              <a
                class="button  primary
    false
    mediumButton
    false
    "
                href="https://www.google.com"
                target="_blank"
              >
                Find out more
              </a>
            </div>
          </div>
        </div>
        <div
          class="column2"
          data-aos="fade-up"
        >
          <div
            class="imgWrap"
          >
            <img
              alt="info-section-img"
              class="img"
              loading="lazy"
              src="/svg.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
